OPHD/States/MapViewState.cpp

```diff
@@ -69,6 +69,43 @@ const std::map<Robot::Type, RobotMeta> RobotMetaTable
 };
 
 
+static NAS2D::Rectangle<int> buildAreaRectFromTile(const Tile& centerTile, int commRange)
+{
+	const NAS2D::Point commAreaStartPoint
+	{
+		std::clamp(centerTile.position().x - commRange, 0, 299),
+		std::clamp(centerTile.position().y - commRange, 0, 149)
+	};
+
+	const NAS2D::Point commAreaEndPoint
+	{
+		std::clamp(centerTile.position().x + commRange, 0, 299),
+		std::clamp(centerTile.position().y + commRange, 0, 149)
+	};
+
+	return NAS2D::Rectangle<int>::Create(commAreaStartPoint, commAreaEndPoint);
+}
+
+
+static void fillCommList(TileList& tileList, TileMap& tileMap, Tile& centerTile, const NAS2D::Rectangle<int> area, int commRange)
+{
+	for (int y = 0; y < area.height; ++y)
+	{
+		for (int x = 0; x < area.width; ++x)
+		{
+			auto& tile = tileMap.getTile({ x + area.x, y + area.y });
+			if (centerTile.distanceTo(&tile) <= commRange)
+			{
+				if (std::find(tileList.begin(), tileList.end(), &tile) == tileList.end())
+				{
+					tileList.push_back(&tile);
+				}
+			}
+		}
+	}
+}
```

Comment:
You may want to consider using `PointInRectangleRange` for this loop:
```cpp
for (const auto point : PointInRectangleRange{area})
{
    auto& tile = tileMap.getTile(point);
    // ...
}
```

Reply:
I didn't know we had that class. There were a lot of features and functions added to NAS2D that I'm unfamiliar with. This might be a good refactor though I'm not sure it would improve performance, only code readability (which... to be fair, since this isn't run per frame may be a non-issue)

Reply:
Efficiency should be identical. It was mainly added to make code easier and more expressive.